pkg/controller/privatelink/actuator/awsactuator/awshubactuator.go

```diff
@@ -374,11 +377,117 @@ func (a *AWSHubActuator) cleanupHostedZone(cd *hivev1.ClusterDeployment, metadat
 }
 
 func (a *AWSHubActuator) ReconcileHostedZoneRecords(cd *hivev1.ClusterDeployment, hostedZoneID string, dnsRecord *actuator.DnsRecord, apiDomain string, logger log.FieldLogger) (bool, error) {
+	hzLog := logger.WithField("hostedZoneID", hostedZoneID)
+	modified := false
+
 	rSet, err := a.recordSet(cd, apiDomain, dnsRecord)
 	if err != nil {
 		return false, errors.Wrap(err, "error generating DNS records")
 	}
 
+	recordsResp, err := a.awsClientHub.ListResourceRecordSets(&route53.ListResourceRecordSetsInput{
```

Comment:
This whole delta exists just so we can tell whether we've changed something?

Ugh, I looked at the API and I can see why that needs to be the case. Gross :(

I guess a `DeepEquals()` isn't sufficient?